server.go

```diff
@@ -1541,6 +1555,73 @@ func (c *createMeasurementsIfNotExistsCommand) addFieldIfNotExists(measurement,
 	return nil
 }
 
+func (s *Server) applyDropSeries(m *messaging.Message) error {
+	var c dropSeriesCommand
+	mustUnmarshalJSON(m.Data, &c)
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	database := s.databases[c.Database]
+	if database == nil {
+		return ErrDatabaseNotFound
+	}
+
+	// Remove from metastore.
+	for _, seriesID := range c.SeriesIDs {
+		series := database.series[seriesID]
+		if series == nil {
+			return fmt.Errorf("series not found for id %d", seriesID)
+		}
+		m := series.measurement
+		if m == nil {
+			return fmt.Errorf("measurement not found for series %d", seriesID)
+		}
+		err := s.meta.mustUpdate(func(tx *metatx) error { return tx.dropSeries(c.Database, m.Name, seriesID) })
+		if err != nil {
+			return err
+		}
+	}
+
+	// Remove shard data
+	for _, rp := range s.databases[c.Database].policies {
+		for _, id := range c.SeriesIDs {
+			err := rp.dropSeries(id)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Delete the database entry.
+	for _, id := range c.SeriesIDs {
+		if !database.removeSeriesFromIndex(id) {
+			return fmt.Errorf("failed to remove series id %d from index", id)
+		}
+	}
+
+	// Delete measurment references to the series
```

Comment:
You shouldn't do this here. You should have a `DropSeries` method on the `Database` struct. The reason is that there is other caching logic contained in there and it'll all need to be updated.

Cleaner to have it own everything. I see there were some references that you missed updating here.